client/my-sites/checkout/get-thank-you-page-url/index.ts

```diff
@@ -233,6 +235,14 @@ export default function getThankYouPageUrl( {
 		return '/';
 	}
 
+	// Gift purchases need to bypass everything below, especially the updateUrlInCookie
+	// redirection to the ecommerce thank you page for ecommerce plan checkouts.
+	// It's possible this might need to be moved further up later if we want to cover gift purchase on Jetpack sites.
+	if ( isGiftPurchase ) {
+		debug( 'gift purchase' );
+		return `/checkout/gift/thank-you/${ siteSlug }`;
```

Comment:
Was wondering about this myself. In what situations would siteSlug be undefined? 

Can we fix that instead of handling a case that shouldn't be happening?

Reply:
Good question. 🤔 It probably is only undefined if this is a siteless purchase. Currently I think there are only three types of siteless purchases: Jetpack checkout, domain-only checkout, and gift renewals. 

For checkout (which is _not_ the only place that calls this function) it is set [here]https://github.com/Automattic/wp-calypso/blob/4219b78b04427af0bbfd63db63b9a2146446b68c/client/my-sites/checkout/checkout-main-wrapper.tsx#L85-L93() and updated [here](https://github.com/Automattic/wp-calypso/blob/4219b78b04427af0bbfd63db63b9a2146446b68c/client/my-sites/checkout/composite-checkout/components/checkout-main.tsx#L142) but originates in the [getSelectedSite selector](https://github.com/Automattic/wp-calypso/blob/4219b78b04427af0bbfd63db63b9a2146446b68c/client/my-sites/checkout/controller.jsx#L88). 

Presumably then it would _always_ be missing in this case since gift purchases [explicitly set the selected site to null](https://github.com/Automattic/wp-calypso/blob/4219b78b04427af0bbfd63db63b9a2146446b68c/client/my-sites/controller.js#L426).

Was it not undefined in your testing? Maybe there's a case where `isGiftPurchase` is true and `siteSlug` is somehow set...

Reply:
See @dylanrevisited's comment below which has a solution.